plugins/module_utils/ldap.py

```diff
@@ -71,7 +78,8 @@ def _connect_to_ldap(self):
             if self.bind_dn is not None:
                 connection.simple_bind_s(self.bind_dn, self.bind_pw)
             else:
-                connection.sasl_interactive_bind_s('', ldap.sasl.external())
+                klass = SASCL_CLASS.get(self.sasl_class, ldap.sasl.external)
```

Comment:
Would it possible to add integration tests for this change?

Reply:
I was thinking about that ... I did find a simple ldap_search integration test but not sure how to set this up for authentication schemes.

Have any pointers?

Reply:
You probably have to modify setup_openldap (https://github.com/ansible-collections/community.general/tree/main/tests/integration/targets/setup_openldap/) to allow both authentication forms. No idea how feasible that is though...

Reply:
(Afterwards you could use the ldap_search tests to use both methods.)